woff, woff2 for regular, italic, bold and bolditalic with scss changes for cayman

maybegeek opened the pull request:
woff and woff2 (latin-extended)

for regular, italic, bold and bolditalic with scss changes for cayman theme.

Perhaps you want to try this out.

thanks,
christoph

maybegeek committed:
rm OpenSans woff add OpenSans woff2

remove OpenSans regular and bold *.woff files. Add OpenSans regular, bold, italic and bold-italic as *.woff2

maybegeek committed:
*.woff added again ... : )

*.woff added again ... : )

maybegeek committed:
scss for *.woff and *.woff2 with bold and boltitalic

scss for *.woff and *.woff2 with bold and boltitalic

maybegeek committed:
fixed forgotten dir in path

`fonts/` directory in path added/fixed

yixuan commented:
Thanks for the patch. Note that the fonts are used by all themes, so I would suggest keeping the existing font files and use consistent file names for other newly added font faces. Also I'm not sure whether it is necessary to include both WOFF and WOFF2. I guess pandoc will include both files in the generated HTML document, making the output unnecessarily large.

maybegeek committed:
rm Open Sans *.woff

SO, css-tricks, caniuse, webfontsgoogle (api, css2) ... going for woff2 only is an viable option.

therefore no *.woff

Naming google-webfont-helper site. Which also shows we have the latin_extended ones.

maybegeek committed:
changed @fontface in all themes

changed the scss to target the new *.woff2 latin_extended Open Sans fonts.

maybegeek commented:
Hi @yixuan,

### *.woff or *.woff2

`*.woff2` should be enough, you are right, we do not need both (`*.woff` and `*.woff2`). Using `*.woff2` is not a progressive enhancement anymore, it is widely used and exclusively used ([https://developers.google.com/fonts/docs/css2](https://developers.google.com/fonts/docs/css2)). So: `woff2` should be enough, I removed the Open Sans `*.woff` previously included.

### naming scheme

I use the naming scheme from [google-webfonts-helper](http://google-webfonts-helper.herokuapp.com/fonts), but any other should do too. I read your comment on keeping existing font files, but that would add more file size mixing `woff` and `woff2`.

### italic, bold-italic and latin-extended

The Open Sans is a widely used font-family ... using the real italic and real bold-italic makes it better.

### changed scss for all themes

I changed the corresponding scss-files to target the 4 `*.woff2` files.

I hope you can test this and hope it will meet your ideas,
thanks

yixuan commented:
Cool! I'm a bit busy these days, but I'll look at the PR more carefully as soon as I get some time.
